### PR TITLE
Feature/compare ingredients

### PR DESCRIPTION
**Pull request by Melizzo**
# PR Description
Please include a summary of the change:
Completed method to be able to find the missing ingredients for a users pantry compared to a specific recipe. 

Commit Emojis:

:tada: Initial commit
:crystal_ball: JS modifications/updates.

If it is a new feature, describe the feature and the iteration of the project spec it addresses.
Pantry iteration

Note any questions or struggles as well.
Struggles: finding the right method to compare the two arrays, and finding the correct data to return - missing ingredients.  

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/pantry.js [hunk 1]
```diff
@@ -1,7 +1,35 @@
 class Pantry {
   constructor(userIngredients) {
-    this.contents = userIngredients;
+    this.pantry = userIngredients;
+    this.neededIngredients = []
   }
+ 
+  // add an array to push in, for ins don't return arrays
+  
+  findMissingIngredients(recipeIngredients)  {
```
Comment (Billwilke42) [inline src/pantry.js:9]: Well done on this method, this was a hard one

File: test/pantry-test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { expect } from 'chai';
+
+import Pantry from '../src/pantry.js';
+import ingredientsData from '../src/data/ingredients.js';
+
+let pantry;
+
```
Comment (Billwilke42) [inline test/pantry-test.js:7]: nice job starting the testing suite!

File: src/pantry.js [hunk 3]
```diff
@@ -2,33 +2,33 @@ class Pantry {
...
+        if(item.ingredient === recipeIngredient.id && !ingredientsWeHave.includes(recipeIngredient)){
+          ingredientsWeHave.push(recipeIngredient)
         }
-      })
+      }) 
     })
-    if (counter === recipeIngredients.length) {
-      return 'You have the ingredients!';
-    }
+    recipeIngredients.forEach(ingredient => {
```
Comment (jeremiahblackol) [inline src/pantry.js:21]: definitely a doozy, glad we made it through!

File: src/pantry.js [hunk 4]
```diff
@@ -2,33 +2,33 @@ class Pantry {
...
-        if (pantryItem.ingredient === recipeIngredient.id) {
-          counter ++;
-        } else {
-          if (!missingIngredients.includes(recipeIngredient)) {
-            missingIngredients.push(recipeIngredient);
-          }
+  
+  findMissingIngredients(recipeIngredients)  {
+    let ingredientsWeHave = []; 
+    const ingredientsWeNeed = []
```
Comment (jeremiahblackol) [inline src/pantry.js:11]: i think you two are right about splitting this up - we'll get there!

**Pull request merged**